test/random.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2021 Matthieu Gautier mgautier@kymeria.fr
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * is provided AS IS, WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, and
+ * NON-INFRINGEMENT.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#include "gtest/gtest.h"
+
+namespace zim {
+  uint32_t randomNumber(uint32_t max);
+};
+
+using namespace zim;
+
+namespace
+{
+TEST(Random, smallMax)
+{
+  for(auto i=0; i<1000; i++) {
+    ASSERT_EQ(randomNumber(0), 0);
+  }
+
+
+  for(auto i=0; i<1000; i++) {
+    auto r = randomNumber(1);
+    ASSERT_TRUE(r>=0 && r<=1);
+  }
+}
+
+TEST(Random, pseudoUnique)
+{
+  auto prev = randomNumber(1000000);
+  for (auto i=0; i<1000; i++) {
+    auto r = randomNumber(1000000);
+    ASSERT_NE(prev, r);
+    prev = r;
+  }
```

Comment:
An alternating sequence 0, 1, 0, 1, ... returned by `randomNumber` will pass this test. On the other hand, in theory this unit-test may occasionally fail (for a good enough random number generator). But that is a universal challenge for creating a good unit test for random number generators. Why don't you look-up for an existing unit test on the internet and adapt it to your function?